profiles/seccomp/kernel_linux.go

```diff
@@ -0,0 +1,75 @@
+package seccomp
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"golang.org/x/sys/unix"
+)
+
+// kernelVersion holds information about the kernel.
+type kernelVersion struct {
+	kernel uint // Version of the kernel (i.e., the "4" in "4.1.2-generic")
+	major  uint // Major revision of the kernel (i.e., the "1" in "4.1.2-generic")
+}
+
+func (k *kernelVersion) String() string {
+	return fmt.Sprintf("%d.%d", k.kernel, k.major)
+}
+
+var (
+	currentKernelVersion *kernelVersion
+	once                 sync.Once
+)
+
+// getKernelVersion gets the current kernel version.
+func getKernelVersion() *kernelVersion {
+	once.Do(func() {
+		var uts unix.Utsname
+		if err := unix.Uname(&uts); err != nil {
+			return
+		}
+		// Remove the \x00 from the release for Atoi to parse correctly
+		currentKernelVersion, _ = parseRelease(string(uts.Release[:bytes.IndexByte(uts.Release[:], 0)]))
+	})
+	return currentKernelVersion
+}
+
+// parseRelease parses a string and creates a kernelVersion based on it.
+func parseRelease(release string) (*kernelVersion, error) {
+	var version = kernelVersion{}
+
+	// We're only make sure we get the "kernel" and "major revision". Sometimes we have
+	// 3.12.25-gentoo, but sometimes we just have 3.12-1-amd64.
+	parsed, err := fmt.Sscanf(release, "%d.%d", &version.kernel, &version.major)
+	if err != nil {
```

Comment:
`|| err != io.EOF`

Reply:
And need to also handle `release == ""` separately

Reply:
Apparently I'm wrong about how Sscanf works. I swear I saw a behavior where EOF detection was needed. Nvm